worlds/hades/Client.py

```diff
@@ -0,0 +1,409 @@
+        await self.get_username()
+        self.tags = set()
+        await self.send_connect()
+
+    async def connection_closed(self):
+        # This is called when the connection is closed (duh!)
+        # This will send the message always, but only process by Styx scribe if actually in game
+        subsume.Send(styx_scribe_send_prefix + "Connection Error")
+        self.is_connected = False
+        self.is_receiving_items_from_connect_package = False
+        await super(HadesContext, self).connection_closed()
+
+    # Do not touch this
+    @property
+    def endpoints(self):
+        if self.server:
+            return [self.server]
+        else:
+            return []
+
+    async def shutdown(self):
+        # What is called when the app gets shutdown
+        subsume.close()
+        await super(HadesContext, self).shutdown()
+
+    # ----------------- Client start up and ending section end  --------------------------------
+
+    # ----------------- Package Management section starts --------------------------------
+
+    def on_package(self, cmd: str, args: dict):
+        # This is what is done when a package arrives.
+        if cmd == "Connected":
+            # What should be done in a connection package
+            self.cache_items_received_names.clear()
+            self.missing_locations_cache = args["missing_locations"]
+            self.checked_locations_cache = args["checked_locations"]
+            self.hades_slot_data = args["slot_data"]
+            if not (self.hades_slot_data["version_check"] == self.polycosmos_version):
+                stringError = "WORLD GENERATED WITH POLYCOSMOS " + self.hades_slot_data["version_check"] \
+                            + " AND CLIENT USING POLYCOSMOS " + self.polycosmos_version + "\n"
+                stringError += "THIS ARE NOT COMPATIBLE"
+                raise Exception(stringError)
+            self.location_name_to_id = self.get_location_name_to_id()
+            if "death_link" in self.hades_slot_data and self.hades_slot_data["death_link"]:
+                asyncio.create_task(self.update_death_link(True))
+                self.deathlink_enabled = True
+            self.is_connected = True
+            self.is_receiving_items_from_connect_package = True 
+
+        if cmd == "RoomInfo":
+            # What should be done when room info is sent.
+            self.seed_name = args["seed_name"]
+        
+        if cmd == "RoomUpdate":
+            if "checked_lodations" in args:
+                collect_locations_cache = ""
+                for location in args["checked_locations"]:
+                    collect_locations_cache += self.location_names.lookup_in_slot(location) + "-"
+                if len(collect_locations_cache) > 0:
+                    collect_locations_cache = collect_locations_cache[:-1]
+                    subsume.Send(styx_scribe_send_prefix + "Locations collected:" + collect_locations_cache)
```

Comment:
```suggestion
        if cmd == "RoomUpdate":
            if "checked_lodations" in args and len(args["checked_locations"]) > 0:
                subsume.Send(styx_scribe_send_prefix + "Locations collected:" + "-".join(
                    self.location_names.lookup_in_slot(location) for location in args["checked_locations"]
                ))
```
Utilizes `str.join`, similar to above.